runtime/near-test-contracts/README.md

```diff
@@ -15,3 +16,14 @@ to the Cargo.toml and use `near_test_contract::rs_contract()`.
 
 If you want to use a contract from an integration test, you can read the wasm file directly from the `./res` directory.
 To populate `./res`, you need to make sure that this crate was compiled.
+
+## Contracts for testing local reproduce transaction execution
+There're three contracts that reflect different state-dependent transaction execution on testnet.
+These contracts are supposed to be deployed on testnet and then obtain a copy of state with rpc, and 
+reproduce its transaction locally with (WIP) near-vm-standalone with custom scripts.
+- simple-state: a simple, one huge state stored for contract, simple contract get/set
+- multiple-state: contract store multiple states under different keys on trie with get/set method
+- cross-contract: cross contract call to read state from another contract or update state of another contract
+
+Above scenarios should cover enough use cases of developer downloading states of testnet contracts, and reproduce it locally.  
```

Comment:
@matklad I could think is single state, multiple state, cross contract read, cross contract write. What other scenarios that have a different workflows in reproducing can you think of?

Reply:
Looking at the https://github.com/near/nearcore/blob/92703b38b6ce131c1bcc69f9cadd6734c089418c/runtime/near-test-contracts/test-contract-rs/src/lib.rs#L7-L159,

I think `random_seed, block_index, block_timestamp, epoch_height, storage_usage` are things we want to specifically test for. I expect that most of those will fall out naturally and won't require any special handlig, but still make sense to double-check. 

With respect to storage, I think doing a `read` of a non-existing key is another case we need to cover.

Reply:
Good points!